### PR TITLE
Allow user to disable auto registration in job

### PR DESCRIPTION
**Pull request by philipvonbargen**
Useage: `--disable-auto-registration`

### CODE REVIEW

File: helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java [hunk 1]
```diff
@@ -378,6 +405,7 @@ public Builder(final String hash, final Parameters parameters) {
       public Resources resources;
       public Map<String, PortMapping> ports;
       public Map<ServiceEndpoint, ServicePorts> registration;
+      public Boolean disableautoRegistration;
```
Comment (rohansingh) [inline helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java]: disable**A**utoRegistration (capitalization)

Comment (philipvonbargen) [inline helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java]: Heh, apparently we did not catch everything while renaming, will update in a bit.

**Comment by davidxia** [comment]
@philipvonbargen I fixed the typos

**Comment by davidxia** [comment]
TODO add registration state like task states

**Comment by rohansingh** [comment]
I think there's still some instances with incorrect casing. Maybe try /[a-z]auto/?

**Comment by philipvonbargen** [comment]
@rohansingh I should have gotten all cases now

**Comment by davidxia** [comment]
@philipvonbargen closing in favor of https://github.com/spotify/helios/pull/398

**Pull request closed without merging**